src/compositions/usePagination.ts

```diff
@@ -0,0 +1,202 @@
+import { Getter } from '@prefecthq/prefect-design'
+import { SubscriptionOptions, useSubscription, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+import { ComputedRef, MaybeRef, Ref, computed, ref, watch } from 'vue'
+import { GLOBAL_API_LIMIT } from '@/compositions/useFilterPagination'
+import { UseSubscriptions, useSubscriptions } from '@/compositions/useSubscriptions'
+import { repeat } from '@/utilities/arrays'
+
+type PaginationFilter = {
+  limit?: number,
+  offset?: number,
+}
+
+// typescript only lets you use any here
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PaginationParameters = [filter?: PaginationFilter, ...any[]]
+type PaginationFetchAction = (...parameters: PaginationParameters) => Promise<unknown[]>
+type PaginationCountAction = (...parameters: PaginationParameters) => Promise<number>
+
+type FetchSubscriptionAction<
+  TFetch extends PaginationFetchAction
+> = (parameters: Parameters<TFetch>[]) => Promise<Awaited<ReturnType<TFetch>>>
+
+export type PaginationOptions = SubscriptionOptions & {
+  mode?: 'page' | 'infinite',
```

Comment:
The `infinite` piece here is in my mind purely part of the UX and not really a difference in the way results are fetched; outside of changing the way results are given back to the caller (flat map + result building) is there anything stopping us from merging these two modes? That would simplify this code in a few places

Reply:
Um `page` returns just the single page of results. Vs `infinite` returns all the pages. 

Also `page` mode starts at page 1 (immediately fetched results) vs `infinite` starts at page 0 and lets the implementation dictate when to fetch results. 

So I don't think they can be combined. Might be missing something

Reply:
Discussed async.